addons/payment_worldline/models/payment_provider.py

```diff
@@ -0,0 +1,201 @@
+        string="Worldline API Secret", required_if_provider='worldline'
+    )
+    worldline_webhook_key = fields.Char(
+        string="Worldline Webhook Key", required_if_provider='worldline'
+    )
+    worldline_webhook_secret = fields.Char(
+        string="Worldline Webhook Secret", required_if_provider='worldline'
+    )
+
+    # === COMPUTE METHODS === #
+
+    def _compute_feature_support_fields(self):
+        """ Override of `payment` to enable additional features. """
+        super()._compute_feature_support_fields()
+        self.filtered(lambda p: p.code == 'worldline').update({
+            'support_tokenization': True,
+        })
+
+    # === BUSINESS METHODS === #
+
+    @api.model
+    def _setup_provider(self, code):
+        """ Override of `payment` to migrate Ogone tokens to Worldline.
+        Remove ogone from account payment method.
+        """
+        super()._setup_provider(code)
+        if code != 'worldline':
+            return
+
+        ogone_providers = self.env['payment.provider'].search([
+            ('code', '=', 'ogone'),
+            ('state', '!=', 'disabled'),
+        ])
+        if ogone_providers:
+            default_worldline = self.env['payment.provider'].search([
+                ('code', '=', 'worldline')
+            ], limit=1)
+
+            for ogone_provider in ogone_providers:
+                company_id = ogone_provider.company_id.id
+
+                # Check if worldline already exists for this company
+                worldline_provider = self.env['payment.provider'].search([
+                    ('code', '=', 'worldline'),
+                    ('company_id', '=', company_id),
+                ], limit=1)
+
+                if not worldline_provider:
+                    worldline_provider = default_worldline.copy({
+                        'company_id': company_id,
+                    })
+
+                # Copy credentials from ogone to worldline
+                worldline_provider.write({
+                    'worldline_pspid': ogone_provider.ogone_pspid,
+                    'worldline_api_key': ogone_provider.ogone_userid,
+                    'worldline_api_secret': ogone_provider.ogone_password,
+                })
+
+                # Transfer tokens from Ogone to Worldline
+                tokens = self.env['payment.token'].search([
+                    ('provider_id', '=', ogone_provider.id)
+                ])
+                if tokens:
+                    tokens.write({'provider_id': worldline_provider.id})
+                    _logger.info(
+                        "Transferred %d token(s) from Ogone to Worldline for company %s",
+                        len(tokens), company_id,
+                    )
+
+            ogone_providers.with_context(skip_token_archival=True).write({'state': 'disabled'})
+
+            # Remove the Ogone payment method
+            self.env['account.payment.method'].search([('code', '=', 'ogone')]).unlink()
```

Comment:
This model define in the `account_payment` module so it will give error when account is not installed. Do we need to unlink this? 🤔